x-pack/plugins/synthetics/server/routes/monitor_cruds/edit_monitor.ts

```diff
@@ -156,6 +159,30 @@ const rollbackUpdate = async ({
   }
 };
 
+export const refreshInlineZip = async (
+  normalizedMonitor: SyntheticsMonitor,
+  previousMonitor: SavedObject<EncryptedSyntheticsMonitorAttributes>,
+  server: SyntheticsServerSetup
+) => {
+  let monitorWithId = {
+    ...normalizedMonitor,
+    [ConfigKey.MONITOR_QUERY_ID]:
+      normalizedMonitor[ConfigKey.CUSTOM_HEARTBEAT_ID] || previousMonitor.id,
+    [ConfigKey.CONFIG_ID]: previousMonitor.id,
+  };
+  if (!!(monitorWithId as BrowserSensitiveSimpleFields)[ConfigKey.SOURCE_INLINE]) {
+    monitorWithId = {
+      ...monitorWithId,
+      [ConfigKey.SOURCE_PROJECT_CONTENT]: await inlineToProjectZip(
+        (monitorWithId as BrowserSensitiveSimpleFields)[ConfigKey.SOURCE_INLINE]!,
```

Comment:
I believe if you extract this to a const, you won't need to add `!`.

Reply:
++ implemented [48255f4](https://github.com/elastic/kibana/pull/176094/commits/48255f412e1b71baa1e83bd99b7e95ebddeb8e91)